ocs_ci/deployment/ibm.py

```diff
@@ -26,206 +30,210 @@ def destroy_ocs(self):
+            if not (
+                p.data["metadata"]["namespace"] == cluster_namespace
+                and p.data["metadata"]["labels"]["app"] == "noobaa"
             )
-            scdevices = localvolume["storageClassDevices"]
-            devicepaths = [sub["devicePaths"] for sub in scdevices]
-            alldevicepaths.append(devicepaths[0][0])
-
-        # STEP 5
-        # delete storage cluster - all of them.
-        scdelete = f"oc delete -n {cluster_namespace} storagecluster --all --wait=true"
-        logger.info(scdelete)
-        out = run_cmd(scdelete)
-        logger.info(out)
-
-        # STEP 6
-        spdelete = f"oc delete project {cluster_namespace} --wait=true --timeout=5m"
-        logger.info(spdelete)
-        out = run_cmd(spdelete)
-        logger.info(out)
-
-        default = "oc project default"
-        logger.info(default)
-        out = run_cmd(default)
-        logger.info(out)
-
-        # STEP 7 & 9
-        nodes = ocp.OCP(kind="node").get().get("items", [])
-        worker_nodes = [
-            node
-            for node in nodes
-            if "cluster.ocs.openshift.io/openshift-storage"
-            in node["metadata"]["labels"]
         ]
-        worker_node_names = [node["metadata"]["name"] for node in worker_nodes]
-        logger.warning(worker_node_names)
-
-        for worker_node in worker_node_names:
-            rookstring = (
-                "oc debug node/"
-                + worker_node
-                + " -- chroot /host rm -rfv /var/lib/rook"
+        pvc.delete_pvcs(rbd_pvcs)
+        cephfs_pvcs = pvc.get_all_pvcs_in_storageclass(constants.CEPHFILESYSTEM_SC)
+
+        # Section 3.1 Step 2
+        # Section 3.3 Step 1
+        # Removing OpenShift Container Platform registry from OpenShift Container Storage
+        registry_conf_name = "configs.imageregistry.operator.openshift.io"
+        registry_conf = ocp.OCP().exec_oc_cmd(f"get {registry_conf_name} -o yaml")
+        if registry_conf["items"][0]["spec"].get("storage", dict()).get("pvc"):
+            patch = dict(spec=dict(storage=dict(emptyDir=dict(), pvc=None)))
+            ocp.OCP().exec_oc_cmd(
+                f"patch {registry_conf_name} cluster --type merge "
+                f"-p '{json.dumps(patch)}'"
+            )
+        # Section 3.3 Step 2
+        pvc.delete_pvcs(cephfs_pvcs)
+
+        # Section 3.1 Step 3
+        try:
+            ocp.OCP().exec_oc_cmd(
+                f"delete -n {cluster_namespace} storagecluster --all --wait=true"
             )
-            logger.info(rookstring)
-            out = run_cmd(rookstring)
-            logger.info(out)
-
-            for devicepath in alldevicepaths:
-                sgdiskstring = (
-                    "oc debug node/"
-                    + worker_node
-                    + " -- chroot /host sgdisk --zap-all "
-                    + devicepath
+        except (CommandFailed, subprocess.TimeoutExpired):
+            pass
```

Comment:
Should we log something here or do we get enough output from `exec_oc_cmd`? If we do decide to add some logging let's also address the other instances of `except` -> `pass`/`continue`.

Reply:
I don't *think* it's necessary to log, since the step after that will delete the namespace and also remove finalizers from several object types including storageclusters if the namespace doesn't delete fully. It just won't be terribly helpful information, and could even be a red herring.